packages/pyright-internal/src/common/realFileSystem.ts

```diff
@@ -405,19 +405,19 @@ class RealFileSystem implements FileSystem {
                 return uri;
             }
 
-            // If it does exist, skip this for symlinks.
-            const path = uri.getFilePath();
-            const stat = fs.lstatSync(path);
-            if (stat.isSymbolicLink()) {
-                return uri;
-            }
-
             // realpathSync.native will return casing as in OS rather than
             // trying to preserve casing given.
-            const realCase = fs.realpathSync.native(path);
+            const realCase = fs.realpathSync.native(uri.getFilePath());

```

Comment:
Just want to confirm that it was intentional that you removed the symlink logic here.

Reply:
I believe this was on purpose. We don't want the original  path, we want the symlinked path, otherwise VS code will open another copy of the file. 

@debonte can correct me if I'm wrong. I believe this was a bug fix that he made.

Reply:
Yes, this is intentional. Here's the explanation from the original pyrx PR:

> RealFileSystem.realCasePath currently only skips real-casing if the file/directory that the input path points to is a symbolic link. But it needs to also skip it if the parent, grandparent, etc directories are symbolic links.
>
> Rather than stat-ing all ancestor directories, I'm checking to see if the real-cased path differs from the original by more than just case. If it does, I'm returning the original path instead.